builtin/providers/aws/resource_aws_elastic_beanstalk_environment.go

```diff
@@ -642,3 +663,26 @@ func dropGeneratedSecurityGroup(settingValue string, meta interface{}) string {
 
 	return strings.Join(legitGroups, ",")
 }
+
+func describeBeanstalkEvents(conn *elasticbeanstalk.ElasticBeanstalk, environmentId string, t time.Time) error {
```

Comment:
@jbardin Thanks for the feedback here, sorry it took a bit of time for me to get around to this.

1.  I agree with changing the name to describeBeanstalkErrors. Originally, I was contemplating having a parameter to set the level of events returned, but I think error is all we really need. 
2. I tried two approaches here, but ran into issues with both. Can you elaborate on how to do this? I tried both returning a slice of errors and a slice of strings. I pushed the slice of errors example to branch on my fork: https://github.com/dharrisio/terraform/commit/950de0f78f1bcc801b00fc64034b530c0c21b5db
What I ran into were type errors about returning `[]error` as type `error`. There was a similar issue with using a slice of strings: `[]string` as type `string` in argument to `fmt.Errorf`.